### PR TITLE
#17901 testing

### PR DESCRIPTION
**Pull request by freddyucv**
Fixing: https://github.com/dotCMS/core/issues/17901#issuecomment-604551395

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 1]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When:Try to create a Group Condition and a Condition in a page's rule with PuBLISH permission over the page
+     * Should: save the rule
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveConditionInRulesPage() throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 2]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        this.addRulesPermission(role, host, true);
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+    }
+
+    private void checkCondition(User user, ConditionGroup conditionGroup, List<Rule> allRules) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid reassigning parameters such as 'allRules'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 3]
```diff
@@ -527,6 +697,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
...
+            final Role role,
+            final Permissionable permissionable,
+            final int permissionPublish) {
+
+        final Permission publishPermission = getPermission(role, permissionable, permissionPublish);
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 4]
```diff
@@ -144,13 +209,75 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
...
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        this.addRulesPermission(role, host, true);
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+    }
+
+    private void checkActions(RuleAction ruleAction, List<Rule> allRules) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'ruleAction' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 5]
```diff
@@ -144,13 +209,75 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
...
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action in a Rule's Page
+     * Should: Save it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveRuleActionInPage() throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 6]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+        addRulesPermission(role, host, true);
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        this.addRulesPermission(role, host, true);
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Local variable 'allRules' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 7]
```diff
@@ -527,6 +697,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
...
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private Permission getPermission(Role role, Permissionable permissionable, int permissionPublish) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 8]
```diff
@@ -16,11 +16,7 @@
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.beans.Permission;
-import com.dotmarketing.business.APILocator;
-import com.dotmarketing.business.CacheLocator;
-import com.dotmarketing.business.PermissionAPI;
-import com.dotmarketing.business.Role;
-import com.dotmarketing.business.Ruleable;
+import com.dotmarketing.business.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 9]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+        final Template template = new TemplateDataGen().nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .nextPersisted();
+
+        addPermission(role, htmlPageAsset, PermissionAPI.PERMISSION_PUBLISH);
+        addRulesPermission(role, host, true);
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        this.addRulesPermission(role, host, true);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: why is this done again? looks like it was done already in line 111

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 10]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
         rulesAPI.saveConditionGroup(conditionGroup, user, false);
 
         List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        checkCondition(user, conditionGroup, allRules);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When:Try to create a Group Condition and a Condition in a page's rule with PuBLISH permission over the page
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: javadoc says `Condition Group` and Condition` but code shows only `Condition Group`

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 11]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Template template = new TemplateDataGen().nextPersisted();
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .folder(folder)
+                .nextPersisted();
+
+        addPermission(role, htmlPageAsset, PermissionAPI.PERMISSION_PUBLISH);        addRulesPermission(role, host, true);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Looks like this line shouldn't be there if you don't want publish perms on the page

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 12]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
         final Rule rule = new RuleDataGen().host(host).nextPersisted();
 
         final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
-        this.addPermission(role, host, true);
+        this.addRulesPermission(role, host, true);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: create a variable for the third parameter to understand what's going on
`boolean addPublishPermsForRules = true;`

Comment (dsilvam) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Or just seperate `addRulesPermission` into two methods with clearer responsabilities:
`addPermsToPublishRules`
`addPermsToReadRules`

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 13]
```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When:Try to create a Group Condition and a Condition in a page's rule without PuBLISH permission over the page
+     * Should: Throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldSaveConditionInRulesPageShouldNotWork() throws DotSecurityException, DotDataException {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: rename method to `shouldNotSaveConditionInRuleOnPage`

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 14]
```diff
@@ -144,13 +209,75 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
...
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action but without PUBLISH permission over the page
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test (expected = DotSecurityException.class)
+    public void shouldSaveRuleActionInPageShouldNotWork() throws DotSecurityException, DotDataException {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: rename method to: `shouldNotSaveRuleActionInRuleOnPage`

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 15]
```diff
@@ -193,13 +320,75 @@ public void shouldDeleteRules() throws DotSecurityException, DotDataException {
...
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User with permission try to delete a rule's page without PUBLISH permission over the page
+     * Should:throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldDeleteRulesInPageShouldNotWork() throws DotSecurityException, DotDataException {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Rename method to: `shouldNotDeleteRuleOnPage`

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 9 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aL115

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 10 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR157

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 11 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aL145

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 12 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR889-R910

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 13 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR136

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 14 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR271

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 15 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR385

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 16]
```diff
@@ -527,6 +712,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
...
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private Permission getPermission(Role role, Permissionable permissionable, int permissionPublish) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 17]
```diff
@@ -82,12 +79,88 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+        addRulesPublishPermissions(role, host);
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Local variable 'allRules' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 18]
```diff
@@ -82,12 +79,88 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When:Try to create a Group Condition and a Condition in a page's rule with PuBLISH permission over the page
+     * Should: save the rule
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveConditionInRulesPage() throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 19]
```diff
@@ -527,6 +712,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
...
+            final Role role,
+            final Permissionable permissionable,
+            final int permissionPublish) {
+
+        final Permission publishPermission = getPermission(role, permissionable, permissionPublish);
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 20]
```diff
@@ -144,13 +217,78 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
...
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action in a Rule's Page
+     * Should: Save it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveRuleActionInPage() throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 21]
```diff
@@ -527,6 +702,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
...
+            final Role role,
+            final Permissionable permissionable,
+            final int permissionPublish) {
+
+        final Permission publishPermission = getPermission(role, permissionable, permissionPublish);
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 22]
```diff
@@ -82,12 +79,85 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When:Try to create a Group Condition and a Condition in a page's rule with PuBLISH permission over the page
+     * Should: save the rule
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveConditionInRulesPage() throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 23]
```diff
@@ -527,6 +702,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
...
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private Permission getPermission(Role role, Permissionable permissionable, int permissionPublish) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 24]
```diff
@@ -82,12 +79,85 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
...
+        addRulesPublishPermissions(role, host);
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Local variable 'allRules' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 25]
```diff
@@ -144,13 +214,75 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
...
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action in a Rule's Page
+     * Should: Save it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveRuleActionInPage() throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 23 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 22 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 24 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Local variable 'allRules' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 25 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 21 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)

**Pull request merged**